src/model.jl

```diff
@@ -215,6 +256,16 @@ function _build_model(m::JuMP.Model; case::Case, storage::Storage,
         storage_map = sparse(storage_bus_idx, sets.storage_idx, 1,
                              sets.num_bus, sets.num_storage)::SparseMatrixCSC
     end
+    # Demand flexibility parameters (if present)
+    bus_flex_amt = _make_bus_flexibility_amount(
+        case, flexibility, start_index, end_index
+    )
+    flexibility_enabled = (bus_flex_amt != zeros(sets.num_bus, interval_length))
+    if flexibility_enabled && (
+        flexibility.duration == nothing || flexibility.duration > interval_length
+    )
+        flexibility.duration = interval_length
```

Comment:
We only support flexibility duration (delta) to be smaller or equal to `interval_length`. This is an implicit assumption which does not reflect in the mathematical formulation. Should put down a warning here to let the user know "any flexibility duration > `interval_length` will be trim down to the `interval_length`? In the future, we may revisit this to implement the work around to carry some extra information to the next interval so that we could remove this limitation.

Reply:
Yeah, I can certainly add that warning (though, to be sure, not an error that stops the simulation, right?). The future functionality that you mention is something that I was considering too, and would likely need to be paired with the look-ahead functionality that I discussed last week (and has also been discussed for energy storage). Once the look ahead is added, we will definitely pass net demand flexibility deviations to the next period (similar to passing energy storage SOC to the next period).

Reply:
The warning regarding demand flexibility duration with respect to interval length has been added. As we discussed, the functionality for including a look-ahead period and passing remaining net demand deviations across intervals will be implemented in a future PR.